shotover-proxy/tests/runner/hotreload_int_tests.rs

```diff
@@ -236,3 +238,182 @@ async fn test_hot_reload_with_zero_connections() {
+    );
+
+    // Verify new certificate client continues to work throughout the transition
+    // Counter starts at 0, this will increment to 1
+    assert_valkey_connection_works(&mut con_new, Some(1), &[("cert_key_0", "cert_value_0")])
+        .unwrap();
+
+    // Wait for multiple drain cycles to observe gradual connection closure
+    tokio::time::sleep(tokio::time::Duration::from_secs(20)).await;
+
+    // Check that more old connections have been drained
+    let mut old_connections_still_working = 0;
+    for con in connections.iter_mut() {
+        if con.get::<_, String>("cert_key_0").is_ok() {
+            old_connections_still_working += 1;
+        }
+    }
+
+    // Should have fewer working connections after additional drain cycles
+    assert!(
+        old_connections_still_working < old_connections_working,
+        "Old connections should be gradually drained. Before: {}, After: {}",
+        old_connections_working,
+        old_connections_still_working
+    );
+
+    // Verify new certificate client is still working perfectly
+    // Counter is now at 1, this will increment to 2
+    assert_valkey_connection_works(&mut con_new, Some(2), &[("cert_key_0", "cert_value_0")])
+        .unwrap();
+
+    // Create another new client to ensure new connections use the new certificate
+    let client_new2 =
+        create_tls_valkey_client_from_certs("127.0.0.1", 6380, "/tmp/shotover-cert-test-new");
+    let mut con_new2 = client_new2.get_connection().unwrap();
+
+    // Verify this new connection also works with new certificate
+    let pong: String = redis::cmd("PING").query(&mut con_new2).unwrap();
+    assert_eq!(pong, "PONG");
+
+    // Verify data is accessible
+    let value: String = con_new2.get("cert_key_5").unwrap();
+    assert_eq!(value, "cert_value_5");
+
+    // Wait for the old shotover to complete draining
+    tokio::time::timeout(
+        Duration::from_secs(120),
+        shotover_old.consume_remaining_events(&[]),
+    )
+    .await
+    .expect("Old shotover should complete drain and shutdown within timeout");
+
+    // Verify all old connections are now closed
+    let mut old_connections_remaining = 0;
+    for con in connections.iter_mut() {
+        if con.get::<_, String>("cert_key_0").is_ok() {
+            old_connections_remaining += 1;
+        }
+    }
+
+    assert_eq!(
+        old_connections_remaining, 0,
+        "All old certificate connections should be drained and closed after full drain cycle"
+    );
+
+    // new certificate connections still work perfectly
+    // Counter is now at 2, this will increment to 3
+    assert_valkey_connection_works(
+        &mut con_new,
+        Some(3),
+        &[
+            ("cert_key_0", "cert_value_0"),
+            ("cert_key_5", "cert_value_5"),
+        ],
+    )
+    .unwrap();
+
+    shotover_new.shutdown_and_then_consume_events(&[]).await;
+
+    // Clean up certificate directories
+    let _ = std::fs::remove_dir_all(cert_dir_old);
```

Comment:
No need to clean up, future test runs need to be able to handle the path existing anyway.